_posts/2022-01-14-code-reviews.md

```diff
@@ -0,0 +1,172 @@
+---
+layout: post
+title: Reviewing finished code is too late - and other tips and motivations for code reviews
+author: lassiautio
+excerpt: >
+  Why should we do code reviews? What makes code reviews even more important than programming? Why and how to shift-left code reviews?
+  This blog post will give answers to these questions and go deeper into the world of code reviews.
+  In the end, there will be also some tips on why code reviews should be small, and why you shouldn't comment too much on code reviews.
+tags:
+ - code review
+---
+
+Code reviews are something we all have been involved, and what we do continuously in our projects. We might have contradictory feelings about them. Often they feel like just bureaucracy that is required by a manager. Sometimes we want someone to give us feedback on our code. When we see poor code, we might think that we should've reviewed the code better.
+
+In this blog post, I will explain why code reviews are important, even more important than programming itself. I will also give the motivation to do code reviews now, not after few days. Some practical tips will be also given to make reviewing easier.
+
+This is a blog post version from my presentation that I had at [DevDay 2021](https://dev.solita.fi/2021/12/13/devday-of-solita.html).
+
+# Reasons to do code reviews
+
+## Learning and sharing knowledge
+
+This might be a little surprising for someones, but reducing bugs or code quality are not the most important reasons to do code reviews. It is sharing the knowledge. Code reviews are meant to share knowledge about the code base and also programming in general.
+
+"Oh, I didn't know it can also be done this way. I will use it next time!" This is something I've said many times when I've reviewed code.
```

Comment:
Have you "said it" or thought to yourself? :D Maybe:

Is something I've thought to myself many times when doing code review.